core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2996,12 +2997,19 @@ else if (operation instanceof MergeInsert && caseColumnNames.isEmpty()) {
                     .map(fieldIndexes::get)
                     .collect(toImmutableList());
 
+            Set<ColumnHandle> nonNullableColumnHandles = metadata.getTableMetadata(session, targetTableHandle).getColumns().stream()
+                    .filter(column -> !column.isNullable() && !column.isHidden())
```

Comment:
Let's remove the hidden filter here. It seems unrelated to nullability. If we later allowed updating hidden columns (not unreasonable), it would be easy to miss updating the code here.

Reply:
Removed.